File: Level1/K번째 수/Solution.js
function solution(array, commands) {
  const answerList = [];
      
  for (let i = 0; i < commands.length; i++) {
    const startIdx = commands[i][0];
    const endIdx = commands[i][1];
    const answerIdx = commands[i][2];

    const temps = [];

    for (let j = startIdx - 1; j <= endIdx - 1; j++) {
      temps.push(array[j]);
    }

    if (answerIdx >= temps.length) {
      answerList.push(temps[answerIdx - temps.length]);
      continue;
    }

    temps.sort((a, b) => a - b);
    answerList.push(temps[answerIdx - 1]);
  }

  return answerList;
}

File: Level1/K번째 수/Solution.java
import java.util.*;

class Solution {
  public int[] solution(int[] array, int[][] commands) {
    List<Integer> answerList = new ArrayList<Integer>();

    for (int i = 0; i < commands.length; i++) {
      int startIdx = commands[i][0];
      int endIdx = commands[i][1];
      int answerIdx = commands[i][2];

      List<Integer> temps = new ArrayList<Integer>();

      for (int j = startIdx - 1; j <= endIdx - 1; j++) {
        temps.add(array[j]);
      }

      if (answerIdx >= temps.size()) {
        answerList.add(temps.get(answerIdx - temps.size()));
        continue;
      }

      Collections.sort(temps);
      answerList.add(temps.get(answerIdx - 1));
    }

    int[] answer = new int[answerList.size()];

    for (int i = 0; i < answer.length; i++) {
      answer[i] = answerList.get(i).intValue();
    }

    return answer;
  }
}